e_ductus/tests/test_e_ductus.py

```diff
@@ -1,52 +1,37 @@
-from unittest import TestCase
-
-# from django.test import TestCase
-from django.contrib.auth.models import User
-from django.http import request
-from django.test import Client, SimpleTestCase
-from django.views.generic import TemplateView
-
-from e_ductus.models import Subject, Course
+from django import urls
 import pytest
-from django.urls import resolve, reverse
-
-from e_ductus.views import HomeView
-
-from e_ductus.views import BaseView
-
-from e_ductus.views import OwnerMixin
-
+from django.contrib.auth import get_user_model
+from django.urls import reverse
 
-class HomeViewTest(TestCase):
-    def test_resolve_to_home_page_view(self):
-        resolver = resolve('/home/')
-        self.assertEqual(resolver.func.view_class, HomeView)
 
-
-class BaseViewTest(TestCase):
-    def test_resolve_to_base_page_view(self):
-        resolver = resolve('/')
-        self.assertEqual(resolver.func.view_class, BaseView)
-
-
-def test_admin(client):
-    assert client.get('/').status_code == 200
+@pytest.mark.parametrize("param", [
+    ("home"),
+    ("base"),
+])
+def test_render_views(client, param):
+    temp_url = urls.reverse(param)
+    resp = client.get(temp_url)
+    assert resp.status_code == 200
 
 
 @pytest.mark.django_db
-def test_models():
-    subject = Subject.objects.create(title="django1", slug="django")
-    assert subject.title == "django1"
-    assert subject.slug == "django"
-
-
-class OwnerMixinTest(SimpleTestCase):
-    
-    class MyView(OwnerMixin, TemplateView):
-        pass
-    
-    def test_owner_mixin(self):
-        my_view = self.MyView()
-        context = my_view.get_context_data()
-        self.assertTrue(context)
-        
+def test_user_login(client, create_test_user, user_data):
+    user_model = get_user_model()
+    assert user_model.objects.count() == 1
+    login_url = urls.reverse("login")
+    resp = client.post(login_url, data=user_data)
+    assert resp.status_code == 302
```

Comment:
Probably worth testing testing which page user is redirected to using `follow=True` and open that page, to check if user really logged in, for example if user name shows up on the page.